pkg/cli/admin/ocpcertificates/trustpurge/trustpurge.go

```diff
@@ -96,24 +141,43 @@ func (r *RemoveOldTrustRuntime) purgeTrustFromConfigMap(cm *corev1.ConfigMap) er
 			return nil
 		}
 
-		caBundle := cm.Data["ca-bundle.crt"]
-		if len(caBundle) == 0 {
+		caBundlePEM := cm.Data["ca-bundle.crt"]
+		if len(caBundlePEM) == 0 {
 			// somebody was faster
 			return nil
 		}
 
-		cm.Data["ca-bundle.crt"] = ""
-		if !r.createdBefore.IsZero() {
-			newBundle, pruned, err := pruneCertBundle(r.createdBefore, caBundle)
-			if err != nil {
-				return fmt.Errorf("cert pruning failed for %s/%s", cmNamespace, cmName)
-			}
-			if !pruned { // old == new
-				return nil
-			}
-			cm.Data["ca-bundle.crt"] = newBundle
+		caCerts, err := certutil.ParseCertsPEM([]byte(caBundlePEM))
+		if err != nil {
+			return fmt.Errorf("failed to parse certificates for %s/%s: %v", cmNamespace, cmName, err)
 		}
 
+		originalSecrets := secretsForBundle(caCerts, r.cachedSecretCerts)
+
+		newBundle, pruned := pruneCertBundle(r.createdBefore, caCerts)
+		if !pruned { // old == new
+			return nil
+		}
+
+		newSecrets := secretsForBundle(newBundle, r.cachedSecretCerts)
+		oldSecretNames, newSecretNames := sets.New[string](), sets.New[string]()
+		for _, s := range originalSecrets {
+			oldSecretNames.Insert(s.namespaceName)
+		}
+		for _, s := range newSecrets {
+			newSecretNames.Insert(s.namespaceName)
+		}
+
+		if NDiffO, ODiffN := newSecretNames.Difference(oldSecretNames), oldSecretNames.Difference(newSecretNames); NDiffO.Len() > 0 || ODiffN.Len() > 0 {
```

Comment:
I guess gaining trust in the new bundle would not be such an issue, this was the initial idea